Sources/GoogleAI/GenerativeModel.swift

```diff
@@ -109,7 +109,7 @@ public final class GenerativeModel {
       response = try await generativeAIService.loadRequest(request: generateContentRequest)
     } catch {
       if let error = error as? RPCError, error.isInvalidAPIKeyError() {
-        throw GenerateContentError.invalidAPIKey(underlying: error)
```

Comment:
Are we losing any additional detail from the server that could be useful by not showing the underlying error?

Reply:
Previously developers could get the following string from the underlying error using `String(describing: underlyingError)`:
`"RPCError(httpResponseCode: 400, message: "API key not valid. Please pass a valid API key.", status: GoogleGenerativeAI.RPCStatus.invalidArgument, details: [GoogleGenerativeAI.ErrorDetails(type: "type.googleapis.com/google.rpc.ErrorInfo", reason: Optional("API_KEY_INVALID"), domain: Optional("googleapis.com"))])"`

Note: `RPCError` is an internal type so developers couldn't do this (before or after this PR) in their own code:
```
do {
  ...
} catch let GenerateContentError.invalidAPIKey(underlying: error as RPCError) {
  ...
}
```

The message _may_ be useful as an associated value but I think most app developers would want to provide their own localized error message (rather than the string returned by the backend).